src/uct/sm/knem/knem_iface.c

```diff
@@ -60,9 +62,15 @@ static UCS_CLASS_INIT_FUNC(uct_knem_iface_t, uct_md_h md, uct_worker_h worker,
                            const uct_iface_params_t *params,
                            const uct_iface_config_t *tl_config)
 {
+    size_t sys_max_iov;
+
     UCS_CLASS_CALL_SUPER_INIT(uct_base_iface_t, &uct_knem_iface_ops, md, worker,
                               tl_config UCS_STATS_ARG(NULL));
     self->knem_md = (uct_knem_md_t *)md;
+
+    sys_max_iov = ucs_max(sysconf(_SC_IOV_MAX), 1);
```

Comment:
according to knem code looks like there is no limit for iovec (which is a bug/security issue). so we can use sysconf(_SC_IOV_MAX), anyway this is standard system check so i suggest to put it in ucs/sys